common/lib/dependabot/git_commit_checker.rb

```diff
@@ -50,7 +50,7 @@ def pinned?
       return true if dependency.version&.start_with?(ref)
 
       # Check the specified `ref` isn't actually a branch
-      !local_upload_pack.match?("refs/heads/#{ref}")
+      !local_upload_pack.match?("\srefs/heads/#{ref}$")
```

Comment:
It took me a minute to reason through this change, specifically the `\s` inclusion. I _think_ having the argument be a regex would make it a bit easier for **me** to parse. What do you think of the following?

````suggestion
      !local_upload_pack.match?(%r{ refs/heads/#{ref}$})